src/esm_runscripts/prepare.py

```diff
@@ -729,3 +730,60 @@ def set_logfile(config):
         "experiment_log_file", log_file_path
     )
     return config
+
+
+def check_config_for_warnings_errors(config):
+
+    triggers = {"error": {"note_function": esm_parser.user_error}}
+
+    # Find conditions to warn (avoid warning more than once)
+    last_jobtype = config["general"].get("last_jobtype", "")
+    isresubmitted = last_jobtype == config["general"]["jobtype"]
+    isinteractive = config["general"].get("isinteractive", "")
+
+    # Only warn if it is an interactive session or while submitted
+    if not isresubmitted or isinteractive:
+        triggers["warning"] = {"note_function": esm_parser.user_note}
+
+    for trigger, trigger_info in triggers.items():
+        warn_error(config, trigger, trigger_info["note_function"])
+
+    return config
+
+def warn_error(config, trigger, note_function):
+
+    if trigger=="warning":
+        sufix_name = f" WARNING"
+    else:
+        sufix_name = f""
+
+    for section, value in config.items():
```

Comment:
Does this go far enough down into nested sections of the config?

Reply:
No, this goes only into choose_blocks, because when they are resolved their content get's into the first indentation. I've done it this way, because it's easier and most importantly, because all our syntax works this way.